diff -Nur valkyrie/src/objects/tool_object.h valkyrien/src/objects/tool_object.h
--- valkyrie/src/objects/tool_object.h	2015-07-13 19:18:50.000000000 +0000
+++ valkyrien/src/objects/tool_object.h	2015-07-13 19:24:56.890873452 +0000
@@ -32,7 +32,7 @@
 #include "toolview/toolview.h"
 #include "utils/vglogreader.h"
 #include "utils/vk_logpoller.h"
-
+#include <unistd.h>
 #include <QList>
 #include <QProcess>
 #include <QStringList>
diff -Nur valkyrie/src/utils/vk_utils.cpp valkyrien/src/utils/vk_utils.cpp
--- valkyrie/src/utils/vk_utils.cpp	2015-07-13 19:18:49.000000000 +0000
+++ valkyrien/src/utils/vk_utils.cpp	2015-07-13 19:25:02.177873357 +0000
@@ -22,6 +22,7 @@
 #include "utils/vk_config.h"        // vkname()
 
 #include <cstdlib>                  // exit, mkstemp, free/malloc, etc
+#include <unistd.h>
 
 #include <QDateTime>
 #include <QFile>
