diff -Nur cargo-0.7.0/cargo-0.7.0/Cargo.toml cargo-0.7.0n/cargo-0.7.0/Cargo.toml
--- cargo-0.7.0/cargo-0.7.0/Cargo.toml	2015-12-09 02:00:31.000000000 +0800
+++ cargo-0.7.0n/cargo-0.7.0/Cargo.toml	2015-12-11 11:30:50.396937637 +0800
@@ -17,37 +17,28 @@
 path = "src/cargo/lib.rs"
 
 [dependencies]
-advapi32-sys = "0.1"
 crates-io = { path = "src/crates-io", version = "0.1" }
-crossbeam = "0.1"
-curl = "0.2"
-docopt = "0.6"
-env_logger = "0.3"
-filetime = "0.1"
-flate2 = "0.2"
-git2 = "0.3"
-git2-curl = "0.3"
-glob = "0.2"
-kernel32-sys = "0.1"
-libc = "0.1"
-libgit2-sys = "0.3"
-log = "0.3"
-num_cpus = "0.2"
-regex = "0.1"
-rustc-serialize = "0.3"
-semver = "0.2.0"
-tar = "0.3"
-term = "0.2"
-time = "0.1"
-toml = "0.1"
-url = "0.2"
-winapi = "0.2"
-
-[dev-dependencies]
-tempdir = "0.3"
-hamcrest = "0.1"
-bufstream = "0.1"
-filetime = "0.1"
+crossbeam = { version = "0.1.5", path = "../crossbeam-0.1.5" }
+curl = { version = "0.2.12", path = "../curl-0.2.12" }
+docopt = { version = "0.6.70", path = "../docopt-0.6.70" }
+env_logger = { version = "0.3.1", path = "../env_logger-0.3.1" }
+filetime = { version = "0.1.6", path = "../filetime-0.1.6" }
+flate2 = { version = "0.2.7", path = "../flate2-0.2.7" }
+git2 = { version = "0.3.2", path = "../git2-0.3.2" }
+git2-curl = { version = "0.3.0", path = "../git2-curl-0.3.0" }
+glob = { version = "0.2.10", path = "../glob-0.2.10" }
+libc = { version = "0.1.12", path = "../libc-0.1.12" }
+libgit2-sys = { version = "0.3.6", path = "../libgit2-sys-0.3.6" }
+log = { version = "0.3.2", path = "../log-0.3.2" }
+num_cpus = { version = "0.2.6", path = "../num_cpus-0.2.6" }
+regex = { version = "0.1.41", path = "../regex-0.1.41" }
+rustc-serialize = { version = "0.3.16", path = "../rustc-serialize-0.3.16" }
+tar = { version = "0.3", path = "../tar-0.3.1" }
+term = { version = "0.2.12", path = "../term-0.2.12" }
+time = { version = "0.1.32", path = "../time-0.1.32" }
+toml = { version = "0.1.23", path = "../toml-0.1.23" }
+url = { version = "0.2.37", path = "../url-0.2.37" }
+semver = { version = "0.2.0", path = "../semver-0.2.0" }
 
 [[bin]]
 name = "cargo"
diff -Nur cargo-0.7.0/cargo-0.7.0/src/crates-io/Cargo.toml cargo-0.7.0n/cargo-0.7.0/src/crates-io/Cargo.toml
--- cargo-0.7.0/cargo-0.7.0/src/crates-io/Cargo.toml	2015-12-09 02:00:31.000000000 +0800
+++ cargo-0.7.0n/cargo-0.7.0/src/crates-io/Cargo.toml	2015-12-11 11:15:47.703953802 +0800
@@ -13,5 +13,5 @@
 path = "lib.rs"
 
 [dependencies]
-curl = "0.2"
-rustc-serialize = "0.3"
+curl = { version = "0.2.12", path = "../../../curl-0.2.12" }
+rustc-serialize = { version = "0.3.16", path = "../../../rustc-serialize-0.3.16" }
