diff -Nur cargo-0.5.0/cargo-0.5.0/Cargo.toml cargo-0.5.0n/cargo-0.5.0/Cargo.toml
--- cargo-0.5.0/cargo-0.5.0/Cargo.toml	2015-09-15 05:48:44.000000000 +0800
+++ cargo-0.5.0n/cargo-0.5.0/Cargo.toml	2015-09-17 16:55:40.858592127 +0800
@@ -17,37 +17,28 @@
 path = "src/cargo/lib.rs"
 
 [dependencies]
-advapi32-sys = "0.1"
-curl = "0.2"
-docopt = "0.6"
-env_logger = "0.3"
-filetime = "0.1"
-flate2 = "0.2"
-git2 = "0.2"
-git2-curl = "0.2"
-glob = "0.2"
-kernel32-sys = "0.1"
-libc = "0.1"
-libgit2-sys = "0.2"
-log = "0.3"
-num_cpus = "0.2"
-regex = "0.1"
+curl = { version = "0.2.11", path = "../curl-0.2.11" }
+docopt = { version = "0.6.70", path = "../docopt-0.6.70" }
+env_logger = { version = "0.3.1", path = "../env_logger-0.3.1" }
+filetime = { version = "0.1.5", path = "../filetime-0.1.5" }
+flate2 = { version = "0.2.7", path = "../flate2-0.2.7" }
+git2 = { version = "0.2.14", path = "../git2-0.2.14" }
+git2-curl = { version = "0.2.4", path = "../git2-curl-0.2.4" }
+glob = { version = "0.2.10", path = "../glob-0.2.10" }
+libc = { version = "0.1.8", path = "../libc-0.1.8" }
+libgit2-sys = { version = "0.2.20", path = "../libgit2-sys-0.2.20" }
+log = { version = "0.3.1", path = "../log-0.3.1" }
+num_cpus = { version = "0.2.6", path = "../num_cpus-0.2.6" }
+regex = { version = "0.1.41", path = "../regex-0.1.41" }
 crates-io = { path = "src/crates-io", version = "0.1" }
-rustc-serialize = "0.3"
-semver = "0.1"
-tar = "0.3"
-term = "0.2"
-threadpool = "0.1"
-time = "0.1"
-toml = "0.1"
-url = "0.2"
-winapi = "0.2"
-
-[dev-dependencies]
-tempdir = "0.3"
-hamcrest = "0.1"
-bufstream = "0.1"
-filetime = "0.1"
+rustc-serialize = { version = "0.3.16", path = "../rustc-serialize-0.3.16" }
+semver = { version = "0.1.20", path = "../semver-0.1.20" }
+tar = { version = "0.3", path = "../tar-0.3.1" }
+term = { version = "0.2.11", path = "../term-0.2.11" }
+threadpool = { version = "0.1.4", path = "../threadpool-0.1.4" }
+time = { version = "0.1.32", path = "../time-0.1.32" }
+toml = { version = "0.1.22", path = "../toml-0.1.22" }
+url = { version = "0.2.37", path = "../url-0.2.37" }
 
 [[bin]]
 name = "cargo"
diff -Nur cargo-0.5.0/cargo-0.5.0/src/crates-io/Cargo.toml cargo-0.5.0n/cargo-0.5.0/src/crates-io/Cargo.toml
--- cargo-0.5.0/cargo-0.5.0/src/crates-io/Cargo.toml	2015-09-15 05:48:44.000000000 +0800
+++ cargo-0.5.0n/cargo-0.5.0/src/crates-io/Cargo.toml	2015-09-17 16:56:27.978871169 +0800
@@ -13,5 +13,5 @@
 path = "lib.rs"
 
 [dependencies]
-curl = "0.2"
-rustc-serialize = "0.3"
+curl = { version = "0.2.11", path = "../../../curl-0.2.11" }
+rustc-serialize = { version = "0.3.16", path = "../../../rustc-serialize-0.3.16" }
