diff -Nur racer-1.0.0/env_logger-0.3.1/Cargo.toml racer-1.0.0n/env_logger-0.3.1/Cargo.toml
--- racer-1.0.0/env_logger-0.3.1/Cargo.toml	2015-08-17 14:43:44.989317530 +0800
+++ racer-1.0.0n/env_logger-0.3.1/Cargo.toml	2015-08-17 14:44:03.015668142 +0800
@@ -11,9 +11,6 @@
 variable.
 """
 
-[dependencies.log]
-version = "0.3"
-path = ".."
-
 [dependencies]
-regex = { version = "0.1.41", path = "../regex-0.1.41" } 
+regex = { version = "0.1.41", path = "../regex-0.1.41" }
+log = { version = "0.3.1", path = "../log-0.3.1" } 
diff -Nur racer-1.0.0/racer-1.0.0/Cargo.toml racer-1.0.0n/racer-1.0.0/Cargo.toml
--- racer-1.0.0/racer-1.0.0/Cargo.toml	2015-08-17 14:43:44.989317530 +0800
+++ racer-1.0.0n/racer-1.0.0/Cargo.toml	2015-08-17 14:44:14.776897710 +0800
@@ -22,6 +22,3 @@
 syntex_syntax = { version = "0.8.0", path = "../syntex_syntax-0.8.0" }
 toml = { version = "0.1.21", path = "../toml-0.1.21" } 
 env_logger = { version = "0.3.1", path = "../env_logger-0.3.1" }
-
-[features]
-nightly = []
