diff --git a/qconf.pro b/qconf.pro
index e8a8b40..97dc789 100644
--- a/qconf.pro
+++ b/qconf.pro
@@ -6,7 +6,9 @@ CONFIG  -= app_bundle
 HEADERS += src/stringhelp.h
 SOURCES += src/stringhelp.cpp src/qconf.cpp
 
-include($$OUT_PWD/conf.pri)
+exists($$OUT_PWD/conf.pri) {
+   include($$OUT_PWD/conf.pri)
+}
 
 contains($$list($$[QT_VERSION]), 4.0.*|4.1.*) {
 	DEFINES += DATADIR=\"$$DATADIR\"
@@ -21,3 +23,12 @@ INSTALLS += target
 libfiles.path = $$DATADIR/qconf
 libfiles.files = $$IN_PWD/conf $$IN_PWD/modules
 INSTALLS += libfiles
+
+isEmpty(CXXFLAGS) {
+    CXXFLAGS=$$(CXXFLAGS)
+}
+
+!isEmpty(CXXFLAGS) {
+    QMAKE_CXXFLAGS_DEBUG = $$CXXFLAGS
+    QMAKE_CXXFLAGS_RELEASE = $$CXXFLAGS
+}
