diff -Nur iphonesdk-utils-2.0/clangwrapper/clangwrapper.cpp iphonesdk-utils-2.0new/clangwrapper/clangwrapper.cpp
--- iphonesdk-utils-2.0/clangwrapper/clangwrapper.cpp	2015-12-13 12:32:44.267789235 +0800
+++ iphonesdk-utils-2.0new/clangwrapper/clangwrapper.cpp	2015-12-13 12:32:36.103789381 +0800
@@ -45,8 +45,8 @@
   }
 
 
-  string sdk_fullpath = "/usr/share/iPhoneOS5.0.sdk";
-  string version = "5.0";
+  string sdk_fullpath = "/usr/share/iPhoneOS9.2.sdk";
+  string version = "9.2";
 
   int should_init = 0;
   if(endWith(string(argv[0]),"wrapper"))
@@ -72,6 +72,9 @@
   // for SDK 4.x, set default arch to armv6
   if(beginWith(version,"4."))
        default_arch = "armv6";
+
+  if(beginWith(version,"9."))
+       default_arch = "arm64";
     
   string command = "clang";
   string caller = argv[0];
