diff -Nur iphonesdk-utils-2.0/clangwrapper/clangwrapper.cpp iphonesdk-utils-2.0n/clangwrapper/clangwrapper.cpp
--- iphonesdk-utils-2.0/clangwrapper/clangwrapper.cpp	2013-09-23 20:13:41.000000000 +0800
+++ iphonesdk-utils-2.0n/clangwrapper/clangwrapper.cpp	2015-12-13 01:19:25.111710872 +0800
@@ -33,7 +33,7 @@
   // (1) detect toolchain.
   string clang = find_command("clang","clang","clang");
   string ldid = find_command("ldid","ldid","ldid");
-  string as = find_command("arm-apple-darwin9-as","arm-apple","as");
+  string as = find_command("arm-apple-darwin11-as","arm-apple","as");
   string target = as.substr(as.find("arm"), as.length()-3-as.find("arm"));
 
   
diff -Nur iphonesdk-utils-2.0/configure.ac iphonesdk-utils-2.0n/configure.ac
--- iphonesdk-utils-2.0/configure.ac	2013-09-23 20:13:41.000000000 +0800
+++ iphonesdk-utils-2.0n/configure.ac	2015-12-13 01:19:42.901710553 +0800
@@ -78,7 +78,8 @@
     LLVM_LIBDIR=`llvm-config --libdir`
     LLVM_LDFLAGS=`llvm-config --ldflags --libs cppbackend|sed 'N;s/\n//'`
     LLVM_LIBS=`llvm-config --libs`
-    LLVM_LIBS="-L${LLVM_LIBDIR} ${LLVM_LIBS} ${LLVM_LDFLAGS}"
+    LLVM_SYSTEM_LIBS=`llvm-config --system-libs`
+    LLVM_LIBS="-L${LLVM_LIBDIR} ${LLVM_LIBS} ${LLVM_SYSTEM_LIBS} ${LLVM_LDFLAGS}"
 
     AC_SUBST(LLVM_CXX_FLAGS)
     AC_SUBST(LLVM_LIBS)
diff -Nur iphonesdk-utils-2.0/genLocalization2/getLocalizedStringFromFile.cpp iphonesdk-utils-2.0n/genLocalization2/getLocalizedStringFromFile.cpp
--- iphonesdk-utils-2.0/genLocalization2/getLocalizedStringFromFile.cpp	2013-09-23 20:13:41.000000000 +0800
+++ iphonesdk-utils-2.0n/genLocalization2/getLocalizedStringFromFile.cpp	2015-12-13 01:19:30.128710782 +0800
@@ -1,6 +1,6 @@
 #include <iostream>
 
-#include "llvm/Config/config.h"
+#include "llvm/Config/llvm-config.h"
 #include "llvm/Support/raw_ostream.h"
 #include "llvm/Support/Host.h"
 
@@ -87,13 +87,20 @@
         *pDiagnosticsEngine,
         fileManager);
 
+
+#if LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 7 
+    const std::shared_ptr<clang::TargetOptions> targetOptions = std::make_shared<clang::TargetOptions>();
+    targetOptions->Triple = triple;//"arm-apple-darwin11";//llvm::sys::getDefaultTargetTriple();
+#else
     clang::TargetOptions targetOptions;
     targetOptions.Triple = triple;//"arm-apple-darwin11";//llvm::sys::getDefaultTargetTriple();
+#endif
+
 #if LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 2 
     clang::TargetInfo *pTargetInfo = 
         clang::TargetInfo::CreateTargetInfo(
             *pDiagnosticsEngine,
-#if LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR == 2
+#if (LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR == 2) || (LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 7)
             targetOptions
 #else
             &targetOptions
@@ -111,7 +118,11 @@
 #endif
                 false);
     clang::HeaderSearch headerSearch(headerSearchOptions,
+#if (LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 7)
+                                     sourceManager,
+#else
                                      fileManager,
+#endif
                                      *pDiagnosticsEngine,
                                      languageOptions,
                                      pTargetInfo);
@@ -153,15 +164,26 @@
         pOpts,
         *pDiagnosticsEngine,
         languageOptions,
+#if ! (LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 7)
         pTargetInfo,
+#endif
         sourceManager,
         headerSearch,
         compInst);
+#if (LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 7)
+   preprocessor.Initialize(*pTargetInfo);
+#endif
+
    clang::FrontendOptions frontendOptions;
+   clang::RawPCHContainerReader Reader;
    clang::InitializePreprocessor(
         preprocessor,
         *pOpts,
+#if (LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 7)
+        Reader,
+#else
         *headerSearchOptions,
+#endif
         frontendOptions);
 
 #else
@@ -185,7 +207,8 @@
 
 
     const clang::FileEntry *pFile = fileManager.getFile(filename);
-    sourceManager.createMainFileID(pFile);
+    sourceManager.setMainFileID( sourceManager.createFileID( pFile, clang::SourceLocation(), clang::SrcMgr::C_User));
+    //sourceManager.createMainFileID(pFile);
     preprocessor.EnterMainSourceFile();
     pTextDiagnosticPrinter->BeginSourceFile(languageOptions, &preprocessor);
 
diff -Nur iphonesdk-utils-2.0/genLocalization2/Makefile.am iphonesdk-utils-2.0n/genLocalization2/Makefile.am
--- iphonesdk-utils-2.0/genLocalization2/Makefile.am	2013-09-23 20:13:41.000000000 +0800
+++ iphonesdk-utils-2.0n/genLocalization2/Makefile.am	2015-12-13 01:19:35.393710688 +0800
@@ -29,8 +29,7 @@
        	$(top_builddir)/libplutil/libplutil.la	\
       	$(top_builddir)/libhelper/libhelper.la	\
        	$(CLANG_LIBS)	\
-	$(LLVM_LIBS)	\
-	-ldl
+	$(LLVM_LIBS)
 
 ios_genLocalization_SOURCES =	\
 	main.cpp	\
