diff -Nur Pinta-1.5/Pinta.Install.proj Pinta-1.5n/Pinta.Install.proj
--- Pinta-1.5/Pinta.Install.proj	2014-05-24 15:50:34.000000000 +0000
+++ Pinta-1.5n/Pinta.Install.proj	2015-06-24 21:53:28.107270805 +0000
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="3.5">
+<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.0">
 
   <!-- Set some paths -->
   <PropertyGroup>
