commit ac3047926cb31e0e842a2bb575e1c37d707d496f
Author: Dan Vrátil <dvratil@redhat.com>
Date:   Mon Mar 30 13:34:25 2015 +0200

    Drop library SONAME
    
    REVIEW: 123148

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f917502..4391fc1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -15,9 +15,4 @@ target_link_libraries(libkeyring-kwallet
     KF5::Wallet
     ${SIGNONEXTENSION_LIBRARIES})
 
-set_target_properties(libkeyring-kwallet PROPERTIES
-    VERSION ${kwalletsignon_VERSION}
-    SOVERSION ${kwalletsignon_VERSION_MAJOR}
-)
-
 install(TARGETS libkeyring-kwallet DESTINATION ${SIGNONEXTENSION_PLUGINDIR})
\ No newline at end of file
