This is part of upstream commit bfd3f88652f4c0eb055776238c1083e4e0916510

diff --git a/src/binwalk/modules/binvis.py b/src/binwalk/modules/binvis.py
index 4835537..a542cff 100644
--- a/src/binwalk/modules/binvis.py
+++ b/src/binwalk/modules/binvis.py
@@ -98,7 +98,7 @@ def _generate_plot_points(self, data_points):
 
         # If the number of data points exceeds the maximum number of allowed data points, use a
         # weighting system to eliminate data points that occur less freqently.
-        if sum(data_points.itervalues()) > self.max_points:
+        if sum(data_points.values()) > self.max_points:
 
             # First, generate a set of weight values 1 - 10
             for i in range(1, 11):
