diff -Nur idea/com/intellij/application/options/InitialConfigurationDialog.form idean/com/intellij/application/options/InitialConfigurationDialog.form
--- idea/com/intellij/application/options/InitialConfigurationDialog.form	2015-12-12 22:08:26.583916072 +0800
+++ idean/com/intellij/application/options/InitialConfigurationDialog.form	2015-12-12 22:08:07.736916410 +0800
@@ -38,67 +38,6 @@
         </constraints>
         <properties/>
       </component>
-      <component id="16dfd" class="javax.swing.JCheckBox" binding="myCreateScriptCheckbox">
-        <constraints>
-          <grid row="7" column="0" row-span="1" col-span="2" vsize-policy="0" hsize-policy="3" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
-        </constraints>
-        <properties>
-          <selected value="false"/>
-          <text value="Enable opening files and projects from the command line"/>
-        </properties>
-      </component>
-      <grid id="a1a85" binding="myCreateScriptPanel" layout-manager="GridLayoutManager" row-count="1" column-count="2" same-size-horizontally="false" same-size-vertically="false" hgap="-1" vgap="-1">
-        <margin top="0" left="0" bottom="0" right="0"/>
-        <constraints>
-          <grid row="8" column="0" row-span="1" col-span="2" vsize-policy="3" hsize-policy="3" anchor="0" fill="3" indent="2" use-parent-layout="false"/>
-        </constraints>
-        <properties/>
-        <border type="none"/>
-        <children>
-          <component id="b0754" class="javax.swing.JLabel">
-            <constraints>
-              <grid row="0" column="0" row-span="1" col-span="1" vsize-policy="0" hsize-policy="0" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
-            </constraints>
-            <properties>
-              <text value="Script path:"/>
-            </properties>
-          </component>
-          <component id="add4a" class="javax.swing.JTextField" binding="myScriptPathTextField">
-            <constraints>
-              <grid row="0" column="1" row-span="1" col-span="1" vsize-policy="0" hsize-policy="6" anchor="8" fill="1" indent="0" use-parent-layout="false">
-                <preferred-size width="150" height="-1"/>
-              </grid>
-            </constraints>
-            <properties/>
-          </component>
-        </children>
-      </grid>
-      <component id="e91fa" class="javax.swing.JCheckBox" binding="myCreateEntryCheckBox">
-        <constraints>
-          <grid row="9" column="0" row-span="1" col-span="2" vsize-policy="0" hsize-policy="3" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
-        </constraints>
-        <properties>
-          <text value="Create desktop entry (integrate in system menu)"/>
-        </properties>
-      </component>
-      <grid id="5ee35" binding="myCreateEntryPanel" layout-manager="GridLayoutManager" row-count="1" column-count="1" same-size-horizontally="false" same-size-vertically="false" hgap="-1" vgap="-1">
-        <margin top="0" left="0" bottom="0" right="0"/>
-        <constraints>
-          <grid row="10" column="0" row-span="1" col-span="2" vsize-policy="3" hsize-policy="3" anchor="0" fill="3" indent="2" use-parent-layout="false"/>
-        </constraints>
-        <properties/>
-        <border type="none"/>
-        <children>
-          <component id="a8b" class="javax.swing.JCheckBox" binding="myGlobalEntryCheckBox">
-            <constraints>
-              <grid row="0" column="0" row-span="1" col-span="1" vsize-policy="0" hsize-policy="3" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
-            </constraints>
-            <properties>
-              <text value="For all  users"/>
-            </properties>
-          </component>
-        </children>
-      </grid>
       <component id="4566" class="com.intellij.ui.components.JBLabel">
         <constraints>
           <grid row="2" column="0" row-span="1" col-span="1" vsize-policy="0" hsize-policy="0" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
diff -Nur idea/com/intellij/application/options/InitialConfigurationDialog.java idean/com/intellij/application/options/InitialConfigurationDialog.java
--- idea/com/intellij/application/options/InitialConfigurationDialog.java	2015-12-12 22:08:26.583916072 +0800
+++ idean/com/intellij/application/options/InitialConfigurationDialog.java	2015-12-12 22:08:07.736916410 +0800
@@ -21,8 +21,6 @@
 import com.intellij.application.options.colors.SimpleEditorPreview;
 import com.intellij.icons.AllIcons;
 import com.intellij.ide.DataManager;
-import com.intellij.ide.actions.CreateDesktopEntryAction;
-import com.intellij.ide.actions.CreateLauncherScriptAction;
 import com.intellij.ide.todo.TodoConfiguration;
 import com.intellij.ide.ui.LafComboBoxRenderer;
 import com.intellij.ide.ui.LafManager;
@@ -66,15 +64,10 @@
   private JPanel myMainPanel;
   private JComboBox myKeymapComboBox;
   private JComboBox myColorSchemeComboBox;
-  private JCheckBox myCreateScriptCheckbox;
-  private JTextField myScriptPathTextField;
-  private JPanel myCreateScriptPanel;
   private JPanel myColorPreviewPanel;
   private JPanel myHeaderPanel;
   private JPanel myFooterPanel;
   private JPanel myExtraOptionsPanel;
-  private JCheckBox myCreateEntryCheckBox;
-  private JCheckBox myGlobalEntryCheckBox;
   private JPanel myCreateEntryPanel;
   private ComboBox myAppearanceComboBox;
   private JLabel myPreferencesLabel;
@@ -163,22 +156,6 @@
     setCancelButtonText("Skip");
     init();
 
-    final boolean canCreateLauncherScript = canCreateLauncherScript();
-    myCreateScriptCheckbox.setVisible(canCreateLauncherScript);
-    myCreateScriptCheckbox.setSelected(canCreateLauncherScript);
-    myCreateScriptPanel.setVisible(canCreateLauncherScript);
-    if (canCreateLauncherScript) {
-      myScriptPathTextField.setText("/usr/local/bin/" + CreateLauncherScriptAction.defaultScriptName());
-    }
-
-    final boolean canCreateDesktopEntry = canCreateDesktopEntry();
-    myCreateEntryCheckBox.setVisible(canCreateDesktopEntry);
-    myCreateEntryCheckBox.setSelected(canCreateDesktopEntry);
-    myCreateEntryPanel.setVisible(canCreateDesktopEntry);
-    if (canCreateDesktopEntry) {
-      myGlobalEntryCheckBox.setSelected(!PathManager.getHomePath().startsWith("/home"));
-    }
-
     myPreferencesLabel.setText("You can use "+ CommonBundle.settingsActionPath() + " to configure any of these settings later.");
 
     Disposer.register(myDisposable, new Disposable() {
@@ -208,14 +185,6 @@
     }
   }
 
-  protected boolean canCreateDesktopEntry() {
-    return CreateDesktopEntryAction.isAvailable();
-  }
-
-  protected boolean canCreateLauncherScript() {
-    return CreateLauncherScriptAction.isAvailable();
-  }
-
   public JComboBox getKeymapComboBox() {
     return myKeymapComboBox;
   }
@@ -411,27 +380,6 @@
     // create default todo_pattern for color scheme
     TodoConfiguration.getInstance().resetToDefaultTodoPatterns();
 
-    final boolean createScript = myCreateScriptCheckbox.isSelected();
-    final boolean createEntry = myCreateEntryCheckBox.isSelected();
-    if (createScript || createEntry) {
-      final String pathName = myScriptPathTextField.getText();
-      final boolean globalEntry = myGlobalEntryCheckBox.isSelected();
-      ProgressManager.getInstance().run(new Task.Backgroundable(project, getTitle()) {
-        @Override
-        public void run(@NotNull final ProgressIndicator indicator) {
-          indicator.setFraction(0.0);
-          if (createScript) {
-            indicator.setText("Creating launcher script...");
-            CreateLauncherScriptAction.createLauncherScript(project, pathName);
-            indicator.setFraction(0.5);
-          }
-          if (createEntry) {
-            CreateDesktopEntryAction.createDesktopEntry(project, indicator, globalEntry);
-          }
-          indicator.setFraction(1.0);
-        }
-      });
-    }
     UIManager.LookAndFeelInfo info = (UIManager.LookAndFeelInfo) myAppearanceComboBox.getSelectedItem();
     LafManagerImpl lafManager = (LafManagerImpl)LafManager.getInstance();
     if (info.getName().contains("Darcula") != (LafManager.getInstance().getCurrentLookAndFeel() instanceof DarculaLookAndFeelInfo)) {
diff -Nur idea/com/intellij/ide/customize/CustomizeIDEWizardDialog.java idean/com/intellij/ide/customize/CustomizeIDEWizardDialog.java
--- idea/com/intellij/ide/customize/CustomizeIDEWizardDialog.java	2015-12-12 22:08:19.188916205 +0800
+++ idean/com/intellij/ide/customize/CustomizeIDEWizardDialog.java	2015-12-12 22:07:44.550916825 +0800
@@ -102,9 +102,6 @@
     if (SystemInfo.isMac) {
       mySteps.add(new CustomizeKeyboardSchemeStepPanel());
     }
-    if (CustomizeDesktopEntryStep.isAvailable()) {
-      mySteps.add(new CustomizeDesktopEntryStep("/UbuntuDesktopEntry.png"));
-    }
 
     PluginGroups pluginGroups = new PluginGroups();
     mySteps.add(new CustomizePluginsStepPanel(pluginGroups));
