diff -Nur swift/utils/build-presets.ini swiftn/utils/build-presets.ini
--- swift/utils/build-presets.ini	2016-06-17 15:37:28.000000000 +0800
+++ swiftn/utils/build-presets.ini	2016-06-17 15:49:55.131973473 +0800
@@ -573,10 +573,6 @@
 build-swift-static-stdlib
 build-swift-stdlib-unittest-extra
 
-# Executes the lit tests for the installable package that is created
-# Assumes the swift-integration-tests repo is checked out
-test-installable-package
-
 # Path to the root of the installation filesystem.
 install-destdir=%(install_destdir)s
 
@@ -588,9 +584,6 @@
 build-subdir=buildbot_linux
 lldb
 release
-test
-validation-test
-long-test
 foundation
 
 dash-dash
