diff -Nur lldb/cmake/modules/LLDBConfig.cmake lldbn/cmake/modules/LLDBConfig.cmake
--- lldb/cmake/modules/LLDBConfig.cmake	2015-12-04 13:08:18.228982758 +0800
+++ lldbn/cmake/modules/LLDBConfig.cmake	2015-12-04 13:08:54.184982115 +0800
@@ -182,6 +182,7 @@
       add_definitions( -DLLDB_PYTHON_HOME="${LLDB_PYTHON_HOME}" )
     endif()
   else()
+    set(Python_ADDITIONAL_VERSIONS 2.7 2.6 2.5)
     find_package(PythonLibs REQUIRED)
   endif()
   
