diff -Nur clang/lib/Driver/ToolChains.cpp clangn/lib/Driver/ToolChains.cpp
--- clang/lib/Driver/ToolChains.cpp	2015-12-04 21:49:24.044422847 +0800
+++ clangn/lib/Driver/ToolChains.cpp	2015-12-04 21:56:33.931528296 +0800
@@ -1427,6 +1427,9 @@
   static const char *const X86_64LibDirs[] = {"/lib", "/lib64"};
   static const char *const X86_64Triples[] = {
       "x86_64-linux-gnu",       "x86_64-unknown-linux-gnu",
+      "x86_64-isoft-linux",     "x86_64-isoft-linux-gnu",
+      "x86_64-everest-linux",   "x86_64-everest-linux-gnu",
+      "x86_64-pure64-linux",    "x86_64-pure64-linux-gnu",
       "x86_64-pc-linux-gnu",    "x86_64-redhat-linux6E",
       "x86_64-redhat-linux",    "x86_64-suse-linux",
       "x86_64-manbo-linux-gnu", "x86_64-linux-gnu",
@@ -1436,6 +1439,9 @@
   static const char *const X86LibDirs[] = {"/lib32", "/lib"};
   static const char *const X86Triples[] = {
       "i686-linux-gnu",       "i686-pc-linux-gnu",     "i486-linux-gnu",
+      "i686-isoft-linux",     "i686-isoft-linux-gnu"
+      "i686-everest-linux",   "i686-everest-linux-gnu",
+      "i686-pure64-linux",    "i686-pure64-linux-gnu",
       "i386-linux-gnu",       "i386-redhat-linux6E",   "i686-redhat-linux",
       "i586-redhat-linux",    "i386-redhat-linux",     "i586-suse-linux",
       "i486-slackware-linux", "i686-montavista-linux", "i686-linux-android",
