diff -Nur clang/lib/AST/ItaniumMangle.cpp clang.abi_tag/lib/AST/ItaniumMangle.cpp
--- clang/lib/AST/ItaniumMangle.cpp	2015-12-06 18:50:14.054191505 +0800
+++ clang.abi_tag/lib/AST/ItaniumMangle.cpp	2015-12-06 18:50:55.554190762 +0800
@@ -31,6 +31,7 @@
 #include "llvm/ADT/StringExtras.h"
 #include "llvm/Support/ErrorHandling.h"
 #include "llvm/Support/raw_ostream.h"
+#include <set>
 
 #define MANGLE_CHECKER 0
 
