diff -Nur checkpolicy-2.3_ori/policy_define.c checkpolicy-2.3/policy_define.c
--- checkpolicy-2.3_ori/policy_define.c	2014-05-07 02:17:56.000000000 +0800
+++ checkpolicy-2.3/policy_define.c	2016-04-22 10:16:14.000000000 +0800
@@ -1574,7 +1574,7 @@
 				goto bad;
 			}
 			class_perm_node_init(perm);
-			perm->class = i + 1;
+			perm->tclass = i + 1;
 			perm->data = datum->s.value;
 			perm->next = avrule->perms;
 			avrule->perms = perm;
@@ -1795,7 +1795,7 @@
 			goto out;
 		}
 		class_perm_node_init(cur_perms);
-		cur_perms->class = i + 1;
+		cur_perms->tclass = i + 1;
 		if (!perms)
 			perms = cur_perms;
 		if (tail)
diff -Nur checkpolicy-2.3_ori/test/dismod.c checkpolicy-2.3/test/dismod.c
--- checkpolicy-2.3_ori/test/dismod.c	2014-05-07 02:17:56.000000000 +0800
+++ checkpolicy-2.3/test/dismod.c	2016-04-22 10:16:30.000000000 +0800
@@ -269,7 +269,7 @@
 
 	cur = avrule->perms;
 	while (cur) {
-		display_id(policy, fp, SYM_CLASSES, cur->class - 1, "");
+		display_id(policy, fp, SYM_CLASSES, cur->tclass - 1, "");
 		cur = cur->next;
 	}
 
@@ -278,7 +278,7 @@
 	fprintf(fp, " ");
 
 	if (avrule->specified & (AVRULE_AV | AVRULE_NEVERALLOW)) {
-		render_access_mask(avrule->perms->data, avrule->perms->class,
+		render_access_mask(avrule->perms->data, avrule->perms->tclass,
 				   policy, fp);
 	} else if (avrule->specified & AVRULE_TYPE) {
 		display_id(policy, fp, SYM_TYPES, avrule->perms->data - 1, "");
