diff -Nur emacs-elixir/elixir-mode.el emacs-elixirn/elixir-mode.el
--- emacs-elixir/elixir-mode.el	2015-12-16 13:36:35.000000000 +0800
+++ emacs-elixirn/elixir-mode.el	2015-12-16 14:31:52.870165426 +0800
@@ -11,7 +11,7 @@
 ;; Created: Mon Nov 7 2011
 ;; Keywords: languages elixir
 ;; Version: 2.2.8
-;; Package-Requires: ((emacs "24") (pkg-info "0.4"))
+;; Package-Requires: ((emacs "24"))
 
 ;; This file is not a part of GNU Emacs.
 
@@ -38,7 +38,6 @@
 
 (require 'easymenu)           ; Elixir Mode menu definition
 (require 'elixir-smie)        ; Syntax and indentation support
-(require 'pkg-info)           ; Display Elixir Mode package version
 
 (defgroup elixir nil
   "Major mode for editing Elixir code."
@@ -473,7 +472,7 @@
 if called interactively, or if SHOW-VERSION is non-nil, otherwise
 just return nil."
   (interactive (list t))
-  (let ((version (pkg-info-version-info 'elixir-mode)))
+  (let ((version "2.2.8"))
     (when show-version
       (message "Elixir-Mode version: %s" version))
     version))
Binary files emacs-elixir/.git/index and emacs-elixirn/.git/index differ
