diff -up sbcl-1.1.13/src/runtime/linux-os.c.personality sbcl-1.1.13/src/runtime/linux-os.c
--- sbcl-1.1.13/src/runtime/linux-os.c.personality	2013-11-01 12:51:32.229193619 -0500
+++ sbcl-1.1.13/src/runtime/linux-os.c	2013-11-01 12:53:57.262562418 -0500
@@ -288,6 +288,10 @@ os_init(char *argv[], char *envp[])
                     setenv("SBCL_IS_RESTARTING", "T", 1);
                     runtime[i] = '\0';
                     execv(runtime, argv);
+                } else {
+                    /* if /proc isn't available (like in chroot builds, like mock), 
+                     * try using execvp with argv[0] instead */
+                    execvp(argv[0], argv);
                 }
             }
             /* Either changing the personality or execve() failed. Either
