diff -up sbcl-1.2.11/generate-version.sh.generate_version sbcl-1.2.11/generate-version.sh
--- sbcl-1.2.11/generate-version.sh.generate_version	2015-04-30 16:57:05.271114866 -0500
+++ sbcl-1.2.11/generate-version.sh	2015-04-30 16:58:00.375330676 -0500
@@ -2,7 +2,7 @@
 
 git_available_p() {
     # Check that (1) we have git (2) this is a git tree.
-    if ( which git >/dev/null 2>/dev/null && git describe >/dev/null 2>/dev/null )
+    if ( test -d .git && which git >/dev/null 2>/dev/null && git describe >/dev/null 2>/dev/null )
     then
         echo "ok"
     else
