diff -Nur qterm-0.5.12/CMakeLists.txt qterm-0.5.12n/CMakeLists.txt
--- qterm-0.5.12/CMakeLists.txt	2015-10-22 23:33:45.696330180 +0800
+++ qterm-0.5.12n/CMakeLists.txt	2015-10-22 23:33:59.884259271 +0800
@@ -23,5 +23,3 @@
   "${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in"
   "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
   IMMEDIATE @ONLY)
-add_custom_target(uninstall
-  "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
