diff -Nur leksah-0.15.1.4/src/IDE/Command.hs leksah-0.15.1.4n/src/IDE/Command.hs
--- leksah-0.15.1.4/src/IDE/Command.hs	2015-08-24 09:17:27.000000000 +0800
+++ leksah-0.15.1.4n/src/IDE/Command.hs	2015-12-10 22:14:58.704793525 +0800
@@ -45,7 +45,7 @@
         pixbufNewFromFile, iconFactoryAddDefault, iconFactoryNew,
         dialogRun, aboutDialogAuthors, aboutDialogWebsite,
         aboutDialogLicense, aboutDialogComments, aboutDialogCopyright,
-        aboutDialogVersion, aboutDialogName, aboutDialogNew, mainQuit,
+        aboutDialogVersion, aboutDialogName, aboutDialogNew, aboutDialogLogo, mainQuit,
         widgetHide, widgetShow, castToWidget, separatorMenuItemNew,
         containerGetChildren, Menu, widgetSetSizeRequest, toolbarSetStyle,
         set, AttrOp(..), castToToolbar, castToMenuBar, uiManagerGetWidget,
@@ -670,9 +670,12 @@
 aboutDialog = do
     d <- aboutDialogNew
     dd <- getDataDir
+    let iconPath = dd </> "pics" </> "leksah.png"
+    logo <- pixbufNewFromFile iconPath
     license <- catch (T.readFile $ dd </> T.unpack (__ "LICENSE")) (\ (_ :: SomeException) -> return "")
     set d [ aboutDialogName := ("Leksah" :: Text)
           , aboutDialogVersion := T.pack $ showVersion version
+          , aboutDialogLogo := Just logo 
           , aboutDialogCopyright := __ "Copyright 2007-2011 Jürgen Nicklisch-Franken, Hamish Mackenzie"
           , aboutDialogComments := __ "An integrated development environement (IDE) for the " <>
                                __ "programming language Haskell and the Glasgow Haskell Compiler"
