diff -Nur smplayer-14.9.0.6690/src/core.cpp smplayer-14.9.0.6690n/src/core.cpp
--- smplayer-14.9.0.6690/src/core.cpp	2015-01-20 22:57:21.000000000 +0000
+++ smplayer-14.9.0.6690n/src/core.cpp	2015-07-17 22:32:44.461745357 +0000
@@ -1811,10 +1811,6 @@
 
 		proc->setOption("ass-font-scale", QString::number(mset.sub_scale_ass));
 
-		if (!pref->mplayer_is_mplayer2) {
-			proc->setOption("flip-hebrew",false); // It seems to be necessary to display arabic subtitles correctly when using -ass
-		}
-
 		if (pref->enable_ass_styles) {
 			if (!pref->force_ass_styles) {
 				// Load the styles.ass file
