diff -Nur gpsd-3.15/SConstruct gpsd-3.15n/SConstruct
--- gpsd-3.15/SConstruct	2015-06-27 15:14:05.997746458 +0000
+++ gpsd-3.15n/SConstruct	2015-06-27 15:13:58.563745909 +0000
@@ -670,6 +670,7 @@
             confdefs.append("#define HAVE_%s 1\n" % f.upper())
         else:
             confdefs.append("/* #undef HAVE_%s */\n" % f.upper())
+    confdefs.append("#define HAVE_CLOCK_GETTIME 1\n")
 
     # Map options to libraries required to support them that might be absent.
     optionrequires = {
