diff -Nur firefox-42.0/browser/components/preferences/in-content/main.js firefox-42.0n/browser/components/preferences/in-content/main.js
--- firefox-42.0/browser/components/preferences/in-content/main.js	2015-10-30 06:17:47.000000000 +0800
+++ firefox-42.0n/browser/components/preferences/in-content/main.js	2015-11-03 09:38:50.413950853 +0800
@@ -527,7 +527,7 @@
       // With 3.0, a new desktop folder - 'Downloads' was introduced for
       // platforms and versions that don't support a default system downloads
       // folder. See nsDownloadManager for details.
-      downloadFolder.label = bundlePreferences.getString("downloadsFolderName");
+      downloadFolder.label = "Downloads";
       iconUrlSpec = fph.getURLSpecFromFile(yield this._indexToFolder(1));
     } else {
       // 'Desktop'
diff -Nur firefox-42.0/toolkit/components/downloads/nsDownloadManager.cpp firefox-42.0n/toolkit/components/downloads/nsDownloadManager.cpp
--- firefox-42.0/toolkit/components/downloads/nsDownloadManager.cpp	2015-10-30 06:17:56.000000000 +0800
+++ firefox-42.0n/toolkit/components/downloads/nsDownloadManager.cpp	2015-11-03 09:40:04.659949523 +0800
@@ -1366,8 +1366,10 @@
   // XDG user dir spec, with a fallback to Home/Downloads
 
   nsXPIDLString folderName;
-  mBundle->GetStringFromName(MOZ_UTF16("downloadsFolder"),
-                             getter_Copies(folderName));
+  // We don't want localized "Downloads" unless XDG is used
+  //mBundle->GetStringFromName(NS_LITERAL_STRING("downloadsFolder").get(),
+  //                           getter_Copies(folderName));
+  folderName.AssignLiteral("Downloads");
 
 #if defined (XP_MACOSX)
   rv = dirService->Get(NS_OSX_DEFAULT_DOWNLOAD_DIR,
