diff -up itcl4.0.3/configure.soname itcl4.0.3/configure
--- itcl4.0.3/configure.soname	2015-05-05 10:57:09.466564679 -0600
+++ itcl4.0.3/configure	2015-05-05 10:58:22.779111760 -0600
@@ -7242,7 +7242,7 @@ fi
 
 	    # TEA specific: use LDFLAGS_DEFAULT instead of LDFLAGS
 	    SHLIB_LD='${CC} -shared ${CFLAGS} ${LDFLAGS_DEFAULT}'
-	    LDFLAGS="$LDFLAGS -Wl,--export-dynamic"
+	    LDFLAGS="$LDFLAGS -Wl,--export-dynamic -Wl,-soname,\${@}"
 	    if test $doRpath = yes; then :
 
 		CC_SEARCH_FLAGS='-Wl,-rpath,${LIB_RUNTIME_DIR}'
