diff --git a/Source/C/GNUmakefile b/Source/C/GNUmakefile
index e0dde3e..bee0059 100644
--- a/Source/C/GNUmakefile
+++ b/Source/C/GNUmakefile
@@ -22,6 +22,9 @@ ifndef ARCH
   ifneq (,$(strip $(filter $(uname_m), x86_64 amd64)))
     ARCH=X64
   endif
+  ifeq ($(patsubst arm%,ARM,$(uname_m)),ARM)
+    ARCH=ARM
+  endif
   ifeq ($(patsubst i%86,IA32,$(uname_m)),IA32)
     ARCH=IA32
   endif
