diff -Nur kate-15.11.90/CMakeLists.txt kate-15.11.90n/CMakeLists.txt
--- kate-15.11.90/CMakeLists.txt	2015-12-03 05:09:42.000000000 +0800
+++ kate-15.11.90n/CMakeLists.txt	2015-12-16 23:20:56.502440912 +0800
@@ -81,9 +81,6 @@
 # let our config.h be found first in any case
 include_directories (BEFORE ${CMAKE_CURRENT_BINARY_DIR})
 
-# kwrite
-ecm_optional_add_subdirectory (kwrite)
-
 # kate application
 ecm_optional_add_subdirectory (kate)
 
diff -Nur kate-15.11.90/doc/CMakeLists.txt kate-15.11.90n/doc/CMakeLists.txt
--- kate-15.11.90/doc/CMakeLists.txt	2015-12-03 05:09:42.000000000 +0800
+++ kate-15.11.90n/doc/CMakeLists.txt	2015-12-16 23:21:00.182441112 +0800
@@ -1,4 +1,3 @@
 # docs
 ecm_optional_add_subdirectory( kate )
 ecm_optional_add_subdirectory( katepart )
-ecm_optional_add_subdirectory( kwrite )
