diff -Nur xdot-0.6/xdot.py xdot-0.6n/xdot.py
--- xdot-0.6/xdot.py	2013-11-05 16:34:39.000000000 +0800
+++ xdot-0.6n/xdot.py	2014-06-17 23:06:46.206412681 +0800
@@ -2047,6 +2047,7 @@
         filter = gtk.FileFilter()
         filter.set_name("Graphviz dot files")
         filter.add_pattern("*.dot")
+        filter.add_pattern("*.gv")
         chooser.add_filter(filter)
         filter = gtk.FileFilter()
         filter.set_name("All files")
